lib/puppet/type/concat_file.rb

```diff
@@ -58,68 +54,98 @@ def exists?
   end
 
   newparam(:order) do
-    desc "Controls the ordering of fragments. Can be set to alphabetical or numeric."
-    defaultto 'numeric'
+    desc "Controls the ordering of fragments. Can be set to :alpha or :numeric."
```

Comment:
I see. The intent of my change was to allow the types to be used more easily directly but the tag normalization was making it tricky. By removing the tag and using the path or title as the target my hope was to get around it.

It *is* very valuable to be able to use the types without the wrappers however. My testing shows a ~10% reduction in total resource count, and ~20% decrease (2.5M -> 2M) in catalog size on one of our concat heavy roles. This adds up real quickly in real-world numbers when you consider an environment of 20-30k hosts which make heavily use of concat, especially on the inventory/puppetdb side of things, since all file content effectively gets duplicated when it gets passed through the wrappers.

Perhaps I can re-factor it to collect on both tag *and* target, title, how does that sound ?